### PR TITLE
Replace mapper by QBMapper

### PR DESCRIPTION
**Pull request by tacruc**
Signed-off-by: Arne Hamann <git@arne.email>

### CODE REVIEW

**Comment by tacruc** [comment]
@nickvergessen Can you confirm that this resolves your NC26 issues?

File: lib/DB/GeophotoMapper.php [hunk 1]
```diff
@@ -12,61 +12,200 @@
...
+	 * @param $fileId
+	 * @param $userId
+	 * @return bool
+	 * @throws \OCP\DB\Exception
+	 */
     public function deleteByFileIdUserId($fileId, $userId) {
-        $sql = 'DELETE FROM `*PREFIX*maps_photos` where `file_id` = ? and `user_id` = ?';
-        return $this->execute($sql, [$fileId, $userId]);
+		try {
+			$entity = $this->findByFileIdUserId( $fileId, $userId);
```
Comment (nickvergessen) [inline lib/DB/GeophotoMapper.php:169]: I wouldn't get just to delete. Just delete directly, if you need true vs. false check `executeStatement() > 0`

File: lib/DB/GeophotoMapper.php [hunk 2]
```diff
@@ -12,61 +12,200 @@
...
+		}
+		if (!is_null($limit)) {
+			$qb->setMaxResults($limit);
+		}
+		return $this->findEntities($qb);
     }
 
+	/**
+	 * @param $fileId
+	 * @return false|void
```
Comment (nickvergessen) [inline lib/DB/GeophotoMapper.php:146]: executeStatement always returns int

```suggestion
	 * @return int
```

File: lib/DB/GeophotoMapper.php [hunk 3]
```diff
@@ -12,61 +12,200 @@
...
+		} catch (DoesNotExistException | MultipleObjectsReturnedException $e) {
+			return false;
+		}
+		$this->delete($entity);
+        return true;
     }
 
+	/**
+	 * @param $userId
+	 * @return bool
```
Comment (nickvergessen) [inline lib/DB/GeophotoMapper.php:179]:
```suggestion
	 * @return int
```

File: lib/DB/GeophotoMapper.php [hunk 4]
```diff
@@ -12,61 +12,200 @@
...
+	 * @param $lng
+	 * @return int
+	 * @throws \OCP\DB\Exception
+	 */
     public function updateByFileId($fileId, $lat, $lng) {
-        $sql = 'UPDATE `*PREFIX*maps_photos` set `lat` = ? , `lng` = ? where `file_id` = ?';
-        return $this->execute($sql, [$lat, $lng, $fileId]);
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->update(
```
Comment (nickvergessen) [inline lib/DB/GeophotoMapper.php:203]: Missing table name

Comment (tacruc) [reply lib/DB/GeophotoMapper.php]: Thanks :+1:

File: lib/DB/GeophotoMapper.php [hunk 5]
```diff
@@ -12,61 +12,199 @@
...
+				$qb->expr()->eq('file_id', $qb->createNamedParameter($fileId, IQueryBuilder::PARAM_STR))
+			);
+
+		return $qb->executeStatement();
     }
 
+	/**
+	 * @param $fileId
+	 * @param $userId
+	 * @return bool
```
Comment (nickvergessen) [inline lib/DB/GeophotoMapper.php:163]:
```suggestion
	 * @return int
```

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**